### PR TITLE
Add info about SMTP to README.md

### PR DESCRIPTION
**Pull request by srandall97**
# Description

I have added instructions for setting up SMTP for both development and production environments.

## Motivation and Context

Previously the documentation didn't give instructions for configuring SMTP so that Quill can actually send email. I wanted to add this documentation so that it would be more clear what is required in order for Quill to be fully functional

## How Has This Been Tested

N/A

## Screenshots (if appropriate)

N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,33 +132,55 @@ mongo --version
...
 ```bash
-npm install
+mkdir db
+mongod --dbpath db --bind_ip 127.0.0.1
 ```
 
-We use `dotenv` to keep track of environment variables, so be sure to stop tracking the `.env` file in Git:
+#### SMTP
+
+This step is optional and is only required if you want to test the email-related functionality of Quill. The easiest option is to use the SMTP server provided by your personal email (Gmail, Outlook, etc.). Look for the documentation about SMTP for your respective email and fill in the values in the `.env` accordingly. Be warned that sending many emails this way is not recommended and this method should only be used for testing.
```
Comment (krubenok) [inline README.md]: One note here for Gmail specifically is that you'll need to enable "less secure apps". https://support.google.com/accounts/answer/6010255?hl=en

File: README.md [hunk 2]
```diff
@@ -133,33 +132,55 @@ mongo --version
...
 
 ```bash
 gulp server
 ```
+
 ### Deploying for your hackathon
+
+#### MongoDB
+
 The database can either be hosted with [MongoDB Atlas](https://www.mongodb.com/cloud/atlas) or on your own server. Atlas will generally be easier to set up and should be the preferred choice unless you are familiar with administering your own server. A guide to setting up Atlas can be found [here](https://docs.atlas.mongodb.com/getting-started/). Note that the URI for the database (which must be specified in `.env`) will be different depending on where your database is hosted.
```
Comment (krubenok) [inline README.md]: Maybe this should be worded as: 

The database can either be hosted with a cloud hosted MongoDB Host such as [MongoDB Atlas](https://www.mongodb.com/cloud/atlas) or on your own server.

There's nothing inherently special about Atlas, it's just the most popular name in cloud-based MongoDB.

File: README.md [hunk 3]
```diff
@@ -133,33 +132,55 @@ mongo --version
...
+
 ### Deploying for your hackathon
+
+#### MongoDB
+
 The database can either be hosted with [MongoDB Atlas](https://www.mongodb.com/cloud/atlas) or on your own server. Atlas will generally be easier to set up and should be the preferred choice unless you are familiar with administering your own server. A guide to setting up Atlas can be found [here](https://docs.atlas.mongodb.com/getting-started/). Note that the URI for the database (which must be specified in `.env`) will be different depending on where your database is hosted. 
 
+#### SMTP
+
+A dedicated SMTP provider is absolutely required if you want Quill to work for your hackathon. There are several providers available (Mailgun, Sendgrid, and others). After setting this up, fill in the `.env` with the values that your provider gives you.
```
Comment (krubenok) [inline README.md]: Maybe link out to the providers we name dropped? 

https://www.mailgun.com
https://sendgrid.com

Also, it's worth noting that both of them have deals they offer free plans via the Github Student pack:
https://sendgrid.com/blog/sendgrid-accelerate-edu/
https://www.mailgun.com/github-students/

**Review by krubenok — APPROVED** [review]
Overall very helpful changes! I just added a few tweaks that I think might be helpful but it's good as-is if you don't think they are worth the change!

File: README.md (hunk 1 above)
Comment (srandall97) [reply README.md]: Right, I will add that

File: README.md (hunk 2 above)
Comment (srandall97) [reply README.md]: Yeah I probably should've looked up more options besides the one you mentioned, I'll add what you've suggested.

**Comment by srandall97** [comment]
I like how this looks a lot better now. Tomorrow I'll finish up my survey and find some people to fill it in.

**Comment by krubenok** [comment]
Can this be merged?

**Pull request merged**